tests/integration/components/bs-dropdown/menu-test.js

```diff
@@ -8,34 +8,83 @@ module('Integration | Component | bs-dropdown/menu', function(hooks) {
+    );
 
-    assert.equal(this.element.querySelector('.dropdown-menu').tagName, 'UL', 'menu is an unordered list (<ul>) by default');
+    assert.equal(
+      this.element.querySelector('.dropdown-menu').tagName,
+      'UL',
+      'menu is an unordered list (<ul>) by default'
+    );
     assert.dom('.dropdown-menu').exists('menu has dropdown-menu class');
     assert.dom('.dropdown-menu').exists('menu has dropdown-menu class');
-    assert.dom('.dropdown-menu').hasClass('dropdown-menu-right', 'menu has dropdown-menu-right class');
+    assert
+      .dom('.dropdown-menu')
+      .hasClass('dropdown-menu-right', 'menu has dropdown-menu-right class');
     assert.dom('.dropdown-menu').hasText('Something');
   });
 
   testBS4('dropdown menu has correct markup', async function(assert) {
-    await render(hbs`{{#bs-dropdown/menu align="right" isOpen=true toggleElement=this.element}}Something{{/bs-dropdown/menu}}`);
+    await render(
+      hbs`{{#bs-dropdown/menu align="right" isOpen=true toggleElement=this.element}}Something{{/bs-dropdown/menu}}`
+    );
 
-    assert.equal(this.element.querySelector('.dropdown-menu').tagName, 'DIV', 'menu is a div (<div>) by default');
+    assert.equal(
+      this.element.querySelector('.dropdown-menu').tagName,
+      'DIV',
+      'menu is a div (<div>) by default'
+    );
     assert.dom('.dropdown-menu').exists('menu has dropdown-menu class');
-    assert.dom('.dropdown-menu').hasClass('dropdown-menu-right', 'menu has dropdown-menu-right class');
+    assert
+      .dom('.dropdown-menu')
+      .hasClass('dropdown-menu-right', 'menu has dropdown-menu-right class');
     assert.dom('.dropdown-menu').hasText('Something');
   });
 
   testBS3('dropdown menu yields item component', async function(assert) {
-    await render(hbs`{{#bs-dropdown/menu toggleElement=this.element isOpen=true as |ddm|}}{{#ddm.item}}Dummy{{/ddm.item}}{{/bs-dropdown/menu}}`);
+    await render(
+      hbs`{{#bs-dropdown/menu toggleElement=this.element isOpen=true as |ddm|}}{{#ddm.item}}Dummy{{/ddm.item}}{{/bs-dropdown/menu}}`
+    );
 
     assert.dom('li').exists({ count: 1 }, 'has item component');
   });
 
   testBS4('dropdown menu yields item component', async function(assert) {
-    await render(hbs`{{#bs-dropdown/menu toggleElement=this.element isOpen=true as |ddm|}}{{#ddm.item}}Dummy{{/ddm.item}}{{/bs-dropdown/menu}}`
+    await render(
+      hbs`{{#bs-dropdown/menu toggleElement=this.element isOpen=true as |ddm|}}{{#ddm.item}}Dummy{{/ddm.item}}{{/bs-dropdown/menu}}`
     );
 
-    assert.dom('.dropdown-item').doesNotExist('has item component with no markup');
+    assert
+      .dom('.dropdown-item')
+      .doesNotExist('has item component with no markup');
+  });
+
+  testBS3('dropdown menu propagates class names', async function(assert) {
+    await render(
+      hbs`{{#bs-dropdown/menu align="right" isOpen=true toggleElement=this.element classNames="custom-class-1 custom-class-2"}}Something{{/bs-dropdown/menu}}`
```

Comment:
I just have this question about this PR: my understanding is that `classNames` is commonly used *inside* the component.js to customize the applied classes, but passing custom class names from the outside is done using the simpler `class` property, as the guides also suggest here: https://guides.emberjs.com/release/components/customizing-a-components-element/#toc_customizing-the-elements-class

At other places I have been passing `class` around like here: https://github.com/kaliber5/ember-bootstrap/blob/263767b46b0f69df6d6c10b764c4ea6148c51dcc/addon/templates/components/common/bs-popover.hbs#L9

What do you think?

Reply:
Good to see this ticket because  need the same functionality.  FWIW I have seen other addons make the distinction between `class` (sets/replaces all) and `extraClass`  (adds to whatever class is already in the component).

Either way, much appreciated, awesome addon 💯

Reply:
@simonihmig good call. I'll make the change.